S3Presigner not working

S3Presigner CDI injection not working.

**ERROR LOG:**

```
Caused by: java.lang.IllegalStateException: The S3Presigner is required but has not been detected/configured.
	at io.quarkus.amazon.s3.runtime.S3ClientProducer.presigner(S3ClientProducer.java:50)
	at io.quarkus.amazon.s3.runtime.S3ClientProducer_Subclass.presigner$$superforward1(Unknown Source)
	at io.quarkus.amazon.s3.runtime.S3ClientProducer_Subclass$$function$$3.apply(Unknown Source)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:54)
	at io.quarkus.arc.runtime.devconsole.InvocationInterceptor.proceed(InvocationInterceptor.java:62)
	at io.quarkus.arc.runtime.devconsole.InvocationInterceptor.monitor(InvocationInterceptor.java:49)
	at io.quarkus.arc.runtime.devconsole.InvocationInterceptor_Bean.intercept(Unknown Source)
	at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:41)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
	at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:32)
	at io.quarkus.amazon.s3.runtime.S3ClientProducer_Subclass.presigner(Unknown Source)
	at io.quarkus.amazon.s3.runtime.S3ClientProducer_ProducerMethod_presigner_67d945a0bddf1bbe94c84fc632b6219048063e1d_Bean.create(Unknown Source)
	at io.quarkus.amazon.s3.runtime.S3ClientProducer_ProducerMethod_presigner_67d945a0bddf1bbe94c84fc632b6219048063e1d_Bean.create(Unknown Source)
	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:96)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:29)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:26)
	at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
	at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
	at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:26)
	at io.quarkus.arc.impl.ClientProxies.getApplicationScopedDelegate(ClientProxies.java:18)
	at io.quarkus.amazon.s3.runtime.S3ClientProducer_ProducerMethod_presigner_67d945a0bddf1bbe94c84fc632b6219048063e1d_ClientProxy.arc$delegate(Unknown Source)
	at io.quarkus.amazon.s3.runtime.S3ClientProducer_ProducerMethod_presigner_67d945a0bddf1bbe94c84fc632b6219048063e1d_ClientProxy.presignGetObject(Unknown Source)
```
	
**Workaround:**
Inject S3Client also 
      or
Use S3Presigner.builder().build();

**Example code:**

     
    @ApplicationScoped
     public class AwsService {

    @ConfigProperty(name = "cloud.storage.s3.presigned-url.timeout")
    int presignedUrlTimeout;

    @Inject
    S3Presigner s3Presigner;

    /**
     * Do not remove the s3client injection as s3Presigner won't initialize without it
     */
    @Inject
    S3Client s3;

    private S3Presigner s3PresignerLocal;

    public AwsService() {
        this.s3PresignerLocal = S3Presigner.builder().build();
    }

    public String getPresignedGetUrl(String bucketName, String key) {
        GetObjectRequest getObjectRequest =
                GetObjectRequest.builder()
                        .bucket(bucketName)
                        .key(key)
                        .build();

        GetObjectPresignRequest getObjectPresignRequest =
                GetObjectPresignRequest.builder()
                        .signatureDuration(Duration.ofMinutes(this.presignedUrlTimeout))
                        .getObjectRequest(getObjectRequest)
                        .build();

        // Generate the presigned request
        PresignedGetObjectRequest presignedGetObjectRequest =
                this.s3Presigner.presignGetObject(getObjectPresignRequest);
       //       PresignedGetObjectRequest presignedGetObjectRequest =  
      //                            this.s3PresignerLocal.presignGetObject(getObjectPresignRequest);
        return presignedGetObjectRequest.url().toString();
    }

quarkus-amazon-services version: 1.0.3
Java Version: openjdk version "17.0.1" 2021-10-19 LTS

--- a/common/deployment/src/main/java/io/quarkus/amazon/common/deployment/AbstractAmazonServiceProcessor.java
+++ b/common/deployment/src/main/java/io/quarkus/amazon/common/deployment/AbstractAmazonServiceProcessor.java
@@ -177,7 +177,7 @@ abstract public class AbstractAmazonServiceProcessor {
                 .map(c -> c.getClientBuilder())
                 .findFirst();
 
-        if (!syncSdkHttpClientBuilder.isPresent() && !asyncSdkAsyncHttpClientBuilder.isPresent()) {
+        if (!syncSdkHttpClientBuilder.isPresent() && !asyncSdkAsyncHttpClientBuilder.isPresent() && presignerBuilder == null) {
             return;
         }
 

